infoblox/resource_infoblox_network_container.go

```diff
@@ -67,19 +86,32 @@ func resourceNetworkContainerCreate(d *schema.ResourceData, m interface{}, isIPv
 		}
 	}
 
-	if cidr == "" || nvName == "" {
-		return fmt.Errorf(
-			"Tenant ID, network view's name and CIDR are required to create a network container")
-	}
-
 	connector := m.(ibclient.IBConnector)
 	objMgr := ibclient.NewObjectManager(connector, "Terraform", tenantID)
-	nc, err := objMgr.CreateNetworkContainer(nvName, cidr, isIPv6, comment, extAttrs)
-	if err != nil {
-		return fmt.Errorf(
-			"creation of IPv6 network container block failed in network view '%s': %s",
-			nvName, err.Error())
+
+	if cidr == "" && parentCidr != "" && prefixLen > 1 {
+		_, err = objMgr.GetNetworkContainer(nvName, parentCidr, isIPv6, nil)
+		if err != nil {
+			return fmt.Errorf(
+				"Allocation of network block within network container '%s' under network view '%s' failed: %s", parentCidr, nvName, err.Error())
```

Comment:
It could be better to wrap the error:

```suggestion
				"Allocation of network block within network container '%s' under network view '%s' failed: %w", parentCidr, nvName, err)
```